lib/new_relic/agent/instrumentation/dynamodb/instrumentation.rb

```diff
@@ -49,10 +49,14 @@ def build_request_with_new_relic(*args)
       @nr_captured_request = yield
     end
 
+    def nr_account_id
+      @nr_account_id ||= NewRelic::Agent::Aws.get_account_id(config)
```

Comment:
the account id will always be the same for a single instance of a client, since it's set up using specific credentials, so we should only need to do the math for that once per client instance

Reply:
I like the idea of memoizing the value to not have to do the conversion every time. How expensive are the `#respond_to?(:access_key_id)` and `#access_key_id` calls? Is there any value in only checking those once and then essentially memoizing `nil` or are they lightweight enough to keep invoking every time and never having the memoization store anything?

Reply:
Updated! https://github.com/newrelic/newrelic-ruby-agent/pull/2684/commits/a58d6cba23522f1a43b212f380839c198d62f5ea